components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java

```diff
@@ -696,6 +700,34 @@ private String getCaptchaParams(String tenantDomain) {
                                 log.debug("Enforcing recaptcha for resend confirmation is not enabled.");
                             }
                         }
+                    } else if (captchaEnabledConfigName.equals(connectorConfig.getName())) {
+                        if (Boolean.parseBoolean(connectorConfig.getValue())) {
+                            Property[] maxFailedConfig = BasicAuthenticatorDataHolder.getInstance()
+                                    .getIdentityGovernanceService()
+                                    .getConfiguration(new String[]{maxFailedAttemptCaptchaConfigName}, tenantDomain);
+                            Property maxFailedProperty = maxFailedConfig[0];
+                            try {
+                                if (Integer.valueOf(maxFailedProperty.getValue()) < failedLoginAttempts) {
+                                    if (log.isDebugEnabled()) {
+                                        log.debug("Number of failed attempts is higher than max failed " +
+                                                "attempts for reCaptcha. Recaptcha will be enforced");
+                                    }
+                                    captchaParams += BasicAuthenticatorConstants.RECAPTCHA_PARAM + "true";
+                                } else {
+                                    if (log.isDebugEnabled()) {
+                                        log.debug("Number of failed attempts is less than max failed " +
+                                                "attempts for reCaptcha. Recaptcha will not be enforced");
+                                    }
```

Comment:
Can't we just end the flow if the attempts have not been exceeded?

Reply:
This is a login scenario. If the number of max failed attempts are higher than some value, we enforce a recaptcha and continue login process. If the max failed attempt is less than the value, we will not enforce the recaptcha and continue the login process. Hence we cannot end the flow in the else condition.

Reply:
What i meant was can't we return the methof from here?